api/v1beta1/hostedcluster_types.go

```diff
@@ -1659,7 +1659,7 @@ type ManagedEtcdStorageSpec struct {
 	//
 	// +optional
 	// +immutable
-	RestoreSnapshotURL []string `json:"restoreSnapshotURL"`
```

Comment:
Why this change?

Reply:
i know... I thought so too. but it actually matters in a kind of dumb way (that I only half understand)

The issue is with how the decoding/encoding of the object works in the webhook. If we don't add `omitempty` here, then the decode/encode logic results in the mutation webhook trying to patch RestoreSnapshotURL to `Null`, then we get an API error during the validation phase about trying to set a `[]string` to a Null value... or something along those lines.